R/post.R

```diff
@@ -203,27 +203,66 @@ is_tweet_length <- function(.x, n = 280) {
   nchar(.x) <= n
 }
 
-upload_media_to_twitter <- function(media, token, alt_text=NULL) {
-  
+#' Uploads media using chunked media endpoint
+#'
+#' @param media Path to media file (image or movie) to upload. 
+#' @param token OAuth token. By default \code{token = NULL}
+#'   fetches a non-exhausted token from an environment
+#'   variable tokens.
+#' @noRd 
+upload_media_to_twitter <- function(media, token, alt_text = NULL) {
   media2upload <- httr::upload_file(media)
+  # equivalent to tools::file_ext
+  file_ext <- function(x) {
+    pos <- regexpr("\\.([[:alnum:]]+)$", x)
+    ifelse(pos > -1L, substring(x, pos + 1L), "")
+  }
+  mediatype <- file_ext(media)
+  stopifnot(mediatype %in% c("jpg", "jpeg", "png", "gif", "mp4"))
   query <- "media/upload"
-  rurl <- paste0(
-    "https://upload.twitter.com/1.1/media/upload.json"
-  )
-  
-  r <- httr::POST(rurl, body = list(media = media2upload), token)
-  
-  out <- httr::content(r)
-
+  rurl <- paste0("https://upload.twitter.com/1.1/media/upload.json")
+  filesize <- file.size(media)
+  gif_no_chunked <- mediatype == "gif" & filesize <= 5*1024*1024
+  if (mediatype %in% c("jpg","jpeg", "png") | gif_no_chunked) {
+    rpost <- httr::POST(rurl, body = list(media = media2upload), token)
+    status_final <- httr::content(rpost)
+  }
+  if (mediatype %in% c("gif", "mp4")) {
+    if (mediatype == "gif") {
+      category <- "tweet_gif"
+    } else {
+      category <- "tweet_video"
+    }
+    init_r <- httr::POST(rurl, body = list(command = "INIT", 
```

Comment:
there's no http response status checking here, I'd use `stop_for_status()` after this POST request unless you know it will always work, which seems unlikely

Reply:
And same comment applies to I think 4 more http requests below with no status checking